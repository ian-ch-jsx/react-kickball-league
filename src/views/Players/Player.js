import { useState, useEffect } from 'react';
import { getPlayersById } from '../../services/players';

export default function Player(props) {
  const [player, setPlayer] = useState({ players: [] });

  useEffect(() => {
    const fetchData = async () => {
      const data = await getPlayersById(props.match.params.id);
      setPlayer(data[0]);
    };
    fetchData();
  }, [props.match.params.id]);
  console.log(player);
  return (
    <div>
      <h1>{player.name}</h1>
      <h2>player info</h2>
      <p>{player.position}</p>
      <p></p>
    </div>
  );
}
