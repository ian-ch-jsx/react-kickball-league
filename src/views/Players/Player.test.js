import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Player from './Player';

test('renders player name', async () => {
  const { container } = render(
    <MemoryRouter>
      <Player match={{ params: { id: 1 } }} />
    </MemoryRouter>
  );
  await screen.findByText('Bennie Jetts');
  expect(container).toMatchSnapshot();
});
