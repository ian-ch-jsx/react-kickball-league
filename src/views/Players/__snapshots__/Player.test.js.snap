// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders player name 1`] = `
<div>
  <div
    class="container"
  >
    <h1>
      Bennie Jetts
    </h1>
    <h2>
      Bridge City Sneakers
    </h2>
    <h3>
      Portland
      , 
      OR
    </h3>
    <h3>
      Position: 
      Pitcher
    </h3>
  </div>
</div>
`;
